packages/wrangler/src/experimental-assets.ts

```diff
@@ -366,6 +368,51 @@ export function processExperimentalAssetsArg(
 	return experimentalAssetsOptions;
 }
 
+/**
+ * Workers assets cannot be used in combination with a few other select
+ * Workers features, such as: legacy assets, sites and tail consumers.
+ *
+ * This function ensures that if any of these mutually exclusive config
+ * keys or command args are used together, an appropriate error is thrown.
+ */
+export function verifyMutuallyExclusiveAssetsArgsOrConfig(
+	args:
+		| Pick<StartDevOptions, "legacyAssets" | "site" | "experimentalAssets">
+		| Pick<DeployArgs, "legacyAssets" | "site" | "experimentalAssets">,
+	config: Config
+) {
+	/*
+	 * - `config.legacy_assets` conflates `legacy_assets` and `assets`
+	 * - `args.legacyAssets` conflates `legacy-assets` and `assets`
+	 */
+	if (
+		(args.experimentalAssets || config.experimental_assets) &&
+		(args.legacyAssets || config.legacy_assets)
+	) {
+		throw new UserError(
+			"Cannot use Experimental Assets and Legacy Assets in the same Worker."
+		);
+	}
+
+	if (
+		(args.experimentalAssets || config.experimental_assets) &&
+		(args.site || config.site)
+	) {
+		throw new UserError(
+			"Cannot use Experimental Assets and Workers Sites in the same Worker."
+		);
+	}
+
```

Comment:
what about this combo?
```
if (
		(args.legacyAssets || config.legacyAssets) &&
		(args.site || config.site)
	) {
		throw new UserError(
			"Cannot use Legacy Assets and Workers Sites in the same Worker."
		);
	}
```

Reply:
this is handled separately outside this fn (see [this comment](https://github.com/cloudflare/workers-sdk/pull/6749/files#r1764903581)). This fn should only handle experimental assets.